client/repolist.c

```diff
@@ -115,12 +117,27 @@ TDNFLoadRepoData(
 
         TDNF_SAFE_FREE_MEMORY(pszRepoFilePath);
         pszRepoFilePath = NULL;
-
         /* may have added multiple repos, go to last one */
         while (*ppRepoNext)
             ppRepoNext = &((*ppRepoNext)->pNext);
     }
 
+    pRepoParsePre = pReposAll;
```

Comment:
```
for (pRepoParsePre = pReposAll;
        pRepoParsePre;
        pRepoParsePre = pRepoParsePre->pNext) {
    for (pRepoParseNext = pRepoParsePre->pNext;
            pRepoParseNext;
            pRepoParseNext = pRepoParseNext->pNext) {
        if (!strcmp(pRepoParsePre->pszId, pRepoParseNext->pszId)) {
            dwError = ERROR_TDNF_DUPLICATE_REPO_ID;
            BAIL_ON_TDNF_ERROR(dwError);
        }
    }
}
```

I would suggest checking of baseurl instead of ID. ID is not a primary key but baseurl is.
If baseurls are same, throw duplicate error.

Reply:
No, that's just plain wrong. The id is the primary key. There is no issue at all if a `baseurl` is duplicated - it's not advisable, but not really an issue.

Having duplicate ids however is a problem.That is why we are doing this.